src/securityPolicy.c

```diff
@@ -624,41 +578,67 @@ security_policy_t policyForSignTxWitness(
 	switch (signTxUsecase) {
 
 	case SIGN_TX_USECASE_ORDINARY_TX: {
-		if (is_valid_input_witness(pathSpec)) {
-			return policyForInputSignTxWitness(pathSpec);
-		}  else if (is_valid_staking_key_witness(pathSpec)) {
-			return policyForStakingKeySignTxWitness(pathSpec);
 
-			#ifdef POOL_OPERATOR_APP
-		} else if (is_valid_pool_cold_key_witness(pathSpec)) {
-			ALLOW(); // for pool retirement certificates
-			#endif // POOL_OPERATOR_APP
+		switch (bip44_classifyPath(pathSpec)) {
 
-		} else {
+		case PATH_WALLET_ADDRESS:
+		case PATH_WALLET_STAKING_KEY:
+		case PATH_POOL_COLD_KEY:
+			if (bip44_isPathReasonable(pathSpec)) {
+				ALLOW();
+			} else {
+				WARN();
```

Comment:
this seems to be more strict than the original logic which allows any valid key - I find it a pointless to warn for unreasonable paths, especially if we don't return the public key - if the tx contains entries actually defined by such keys we already warn the user si warning them again during signing seems kind of redundant to me

Reply:
ah I see, the "WARN()" directive was embedded in the functions, so it's not really different, maybe apart from cold keys

Reply:
The problem with witnesses is that they can be totally unrelated to the tx (no way to check). So whenever something suspicious occurs, it is better to warn the user.